clearwater-diags-monitor/usr/share/clearwater/bin/gather_diags_utils

```diff
@@ -0,0 +1,95 @@
+#!/bin/sh
+
+# @file gather_diags_verification
+#
+# Copyright (C) Metaswitch Networks 2017
+# If license terms are provided to you in a COPYING file in the root directory
+# of the source code repository by which you are accessing this code, then
+# the license outlined in that COPYING file applies to your use.
+# Otherwise no rights are granted except for those provided to you by
+# Metaswitch Networks in a separate written agreement.
+
+
+# Functions that can be called to:
+# - Check the diags monitor process is actually running
+# - Get the current CPU idle percentage (rounded down)
+# - Check the user is happy to run a CPU intensive operation
+
+
+# Check for a diags monitor process. Alert, and attempt restart, if none found
+verify_process_running ()
+{
+  process_running=$(pgrep -f "/usr/share/clearwater/bin/clearwater_diags_monitor" > /dev/null ; echo $?)
+  if [ $process_running != 0 ]
+  then
+    echo "Warning: No clearwater-diags-monitor process found running. Attempting restart"
+    service clearwater-diags-monitor start
+    if [ $? != 0 ]
+    then
+      echo "Error: Unable to start clearwater-diags-monitor. Aborting operation"
+      exit 1
+    else
+      # Double check that the process stays up after the restart
+      sleep 5
+      process_running=$(pgrep -f "/usr/share/clearwater/bin/clearwater_diags_monitor" > /dev/null ; echo $?)
+      if [ $process_running != 0 ]
+      then
+        echo "Error: Clearwater-diags-monitor not running after restart. Aborting operation"
+        exit 1
+      fi
+      echo "Successfully started clearwater-diags monitor"
+    fi
+  fi
+}
+
+# Required idle CPU for triggering gathering diagnostics
+MIN_IDLE_CPU_FOR_GATHER=40
+
+# Gets the idle CPU averaged over a 10s period.
+get_idle_cpu ()
+{
+  # Use sar to get the CPU usage over 10s, find the summary ("all") line, and
+  # then grab the idle value (last field on the row), removing any fractional
+  # part.
+  sar -P ALL 10 1 |
+  grep "Average:  *all" |
+  sed -e 's/^.* //g' |
+  sed -e 's/\..*//g'
+}
+
+# Asks a user to either wait for idle CPU, or re-run with --force
+check_and_wait_cpu_idle ()
+{
+  echo "This system is currently running at a higher CPU usage than the configured threshold, with less than $MIN_IDLE_CPU_FOR_GATHER % idle CPU"
+  echo "The gather_diags process will wait until we have enough idle CPU, or it can be re-run with the --force flag to skip CPU usage checks"
+  echo "Enter 'y' to continue and wait for idle CPU, or any other input to exit"
```

Comment:
Should we allow y and Y?

Reply:
i mean, it says 'y', not 'Y'. You know how to do a case insensitive match in bash? Feels like more effort than it's worth really